### ISSUE TITLE
Pause attacks on stun

### ISSUE THREAD

**Issue by ohrrpgce-bugbot** [ohrrpgce/ohrrpgce#587]
[bz#587]

When you hit a hero or enemy with a stun attack, they continue their current chain of attacks. Of course, you will often want this for complicated effects, so perhaps we should add an attack bitset to specify whether chains are broken or not, or whether they resume when the character's stun expires. I think the default should be to break chains.

Note that this also happens if a hero or enemy is stunned after choosing an attack, but before starting it, where it really looks bad.

From: @rversteegen
Severity:     feature request
Duplicates:   bz⁠#237

**Comment by ohrrpgce-bugbot** [comment]
Comment author: @bob-the-hamster

There is already a bitset for this. Just turn on "Cancel target's attack" for the stun attack.

See also bug bz⁠#237

**Comment by ohrrpgce-bugbot** [comment]
Comment author: @rversteegen

Opps, what happened there? My finger slipped...

Talking to Rya further with this, I'm reopening to suggest a way to have stun pause attacks - this is, pause the attack delay - because cancelling attacks isn't always desired.

**Comment by ohrrpgce-bugbot** [comment]
Comment author: **msw188**

I agree, and I think this is a serious issue.  Right now, if a character chooses an attack with a delay, and then is stunned before committing the attack, his turn is lost.  Not only does he NOT commit his delayed attack, he also does not get to choose another one, and his battle-ready-meter is put back to empty.  This is WITHOUT the "Cancel Target's Attack" bitset.

It seems to me that what I've described is what should happen with that bitset turned on, while having it turned off ought to allow the delayed attack to execute after the stun has been removed.

**Comment by ohrrpgce-bugbot** [comment]
Comment author: @rversteegen


> Talking to Rya further with this, I'm reopening to suggest a way to have stun
> pause attacks - this is, pause the attack delay - because cancelling attacks
> isn't always desired.

I don't understand what I meant when I said this. I think I was confused. Now I'm even more confused. I don't see a suggestion for something more than "Cancel Target's Attack"; I don't think I should have reopened this bug.

I think it would be simplest to break msw's clearly explained bug report to a fresh bug.

\*\*\* This bug has been marked as a duplicate of bug bz⁠#237 \*\*\*
